src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -31,15 +32,14 @@ internal sealed class AspNetCoreDiagnosticObserver : DiagnosticObserver
         private static readonly int PrefixLength = "Microsoft.AspNetCore.".Length;
 
         private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<AspNetCoreDiagnosticObserver>();
-
-        private static readonly PropertyFetcher HttpRequestInStartHttpContextFetcher = new PropertyFetcher("HttpContext");
-        private static readonly PropertyFetcher HttpRequestInStopHttpContextFetcher = new PropertyFetcher("HttpContext");
-        private static readonly PropertyFetcher UnhandledExceptionExceptionFetcher = new PropertyFetcher("Exception");
-        private static readonly PropertyFetcher BeforeActionHttpContextFetcher = new PropertyFetcher("httpContext");
-        private static readonly PropertyFetcher BeforeActionActionDescriptorFetcher = new PropertyFetcher("actionDescriptor");
-
         private readonly Tracer _tracer;
 
+        private string hostingHttpRequestInStartEventKey;
+        private string mvcBeforeActionEventKey;
+        private string hostingUnhandledExceptionEventKey;
+        private string diagnosticsUnhandledExceptionEventKey;
+        private string hostingHttpRequestInStopEventKey;
```

Comment:
Style nit: The convention in this repo is to prefix the name of private instance fields with an underscore. 

```suggestion
        private string _hostingHttpRequestInStartEventKey;
        private string _mvcBeforeActionEventKey;
        private string _hostingUnhandledExceptionEventKey;
        private string _diagnosticsUnhandledExceptionEventKey;
        private string _hostingHttpRequestInStopEventKey;
```

I don't have a personal preference either way, as long as we are consistent throughout. (I'm surprised StyleCop isn't complaining about this 😅 )